lib/celluloid/future.rb

```diff
@@ -21,12 +21,36 @@ def self.new(*args, &block)
 
     attr_reader :address
 
-    def initialize
+    def initialize(&block)
       @address = Celluloid.uuid
       @mutex = Mutex.new
       @ready = false
       @result = nil
       @forwards = nil
+      @cancelled = false
```

Comment:
How did these changes get in there? These already exist on the `0.17.0-prerelease` branch that the `GSoC/2015` branch came from.